### ISSUE TITLE
Edge cases for the Windows High Contrast mode focus style

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5244]
Follow up to https://github.com/WordPress/gutenberg/pull/5138

Thanks to the brilliant solution from @jasmussen now the focus style for Windows High Contrast mode is implemented with a simple solution, a 2 pixels transparent outline which is visible only in High Contrast mode.

I've noticed a few places where it could be improved:

1.
The "Settings" button: for testing purposes there's no need to switch to Windows High Contrast mode; in the screenshot below I've made the outline red to better illustrate the issue. The outline works when the sidebar is closed:

<img width="155" alt="screen shot 2018-02-25 at 14 20 01" src="https://user-images.githubusercontent.com/1682452/36641914-e40657d2-1a37-11e8-86bb-0ae9212b1499.png">

But when the button is toggled, it gets an `outline: none` from a CSS rule with higher specificity, so the focus style isn't visible when the sidebar is open:
<img width="157" alt="screen shot 2018-02-25 at 14 19 41" src="https://user-images.githubusercontent.com/1682452/36641922-fd658874-1a37-11e8-9a11-686bc1ebadbb.png">

2.
The "Publish" button
It inherits the focus style from core so it uses box-shadow. Focus style is not visible in Windows High Contrast mode, wondering if it should be fixed in core or patched in Gutenberg.

3
The inserter panel tabs
They use just box-shadow, so there's no focus indication in Windows High Contrast mode.
